contracts/reverseRegistrar/profiles/L2ReverseResolverBase.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.4;
+
+import "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import "../../resolvers/profiles/IVersionableResolver.sol";
+
+abstract contract L2ReverseResolverBase is ERC165 {
+    mapping(bytes32 => uint64) internal recordVersions;
+    event VersionChanged(bytes32 indexed node, uint64 newVersion);
+    bytes32 public immutable L2_REVERSE_NODE;
+
+    bytes32 constant lookup =
+        0x3031323334353637383961626364656600000000000000000000000000000000;
+
+    function isAuthorised(address addr) internal view virtual returns (bool);
+
+    constructor(bytes32 l2ReverseNode) {
+        L2_REVERSE_NODE = l2ReverseNode;
+    }
+
+    modifier authorised(address addr) virtual {
+        require(isAuthorised(addr));
+        _;
+    }
+
+    /**
+     * Increments the record version associated with an ENS node.
+     * May only be called by the owner of that node in the ENS registry.
+     * @param addr The node to update.
+     */
+    function clearRecords(address addr) public virtual authorised(addr) {
```

Comment:
Can you add test for this function as the implementation defers a lot from the L1 equivalent.

Reply:
The implementation code of `clearRecords` and `clearRecordsWithSignature` are identical (only difference in modifier). Can you refactor to avoid code duplication?